src/components/checkout-button.ts

```diff
@@ -1,127 +1,129 @@
-  shopAddress = '';
+  static get styles() {
+    return [
+      themeStyles,
+      buttonStyles,
+      typographyStyles
+    ]
+  }
 
+  /**
+   * Name of the product to buy
+   */
   @property({type: String}) 
   label = '';
-
+  
   protected get hasLabel(): Boolean {
     return this.label.trim().length > 0
   }
-
+  
+  /**
+   * Name of the product to buy
+   */
   @property({type: String}) 
-  name = null;
-
-  @property({type: String, attribute: 'button-class'}) 
-  buttonClass = 'nq-button';
+  name = '';
   
+  protected get hasName(): Boolean {
+    return this.name.trim().length > 0
+  }
+  
+  /**
+   * Price of the product to buy
+   */
   @property({type: Number}) 
   price = 0;
-  
-  @property({type: SubmitState}) 
-  submitState = SubmitState.NotSubmitted;
-
-  // error messages
-  errors: String[] = []
 
-  // Nimiq Hub Api
-  hubApi: HubApi = new HubApi(process.env.HUB_API)
+  /**
+   * Flag to inverse button style
+   */
+  @property({type: Boolean}) 
+  inverse = false;
 
-  // Storage
-  // TODO(sectore): Put 'repo' into storage
-  storage: ShopStorage = process.env.production ? 
-    new IpfsShopStorage('') : new DummyShopStorage('')
-
-  protected get hasErrors(): Boolean {
-    return this.errors.length > 0
-  }
-
-  connectedCallback() {
-    super.connectedCallback()
-    console.log(`checkout button for ${this.name} connected`)
-    this.validate()
+  protected get buttonInverseClass(): String {
+    return (this.inverse) ? 'inverse' : ''
   }
 
-  attributeChangedCallback(name, oldval, newval) {
-    // console.log('attribute change: ', name, newval);
-    super.attributeChangedCallback(name, oldval, newval);
+  /**
+   * Flag to style a small button size
+   */
+  @property({type: Boolean}) 
+  small = false;
+
+  /**
+   * Flag to style a pill button
+   */
+  @property({type: Boolean}) 
+  pill = false;
+  
+  protected get buttonClass(): String {
+    return (this.small) 
+      ? 'nq-button-s' : (this.pill) 
+        ? 'nq-button-pill' 
+        : 'nq-button'
   }
-
-  protected validate():void {
-    this.errors = []
-    if (this.price <= 0) {
-      this.errors.push(`Missing or invalid price`)
-    }
-    if (this.name == null || this.name == '') {
-      this.errors.push(`Missing product name`)
-    }
-    // TODO: Add a proper validation of an Nimiq address
-    if (this.shopAddress == '') {
-      this.errors.push(`Missing or invalid address ${this.shopAddress}`)
+  
+  /**
+   * Flag to disable the button
+   */
+  @property({type: Boolean}) 
+  disable = false;
+
+  /**
+   * Button color class
+   */
+  @property({type: String}) 
+  buttonColor = '';
+  
+  protected get buttonColorClass(): String {
+    let clazz: String;
+    switch (this.buttonColor) {
+      case 'light-blue':
+      case 'green':
+      case 'orange':
+      case 'red':
+      case 'gold':
+        clazz = this.buttonColor
+        break;
+      default: 
+        clazz = ''
```

Comment:
I'm always a big fan of sharing a similar coding style.
I'd have gone for something (pseudo code)
```return `['green', 'orange', ...].contains(this.buttonColor) ? this.buttonColor: ''```
Wdyt? How do you feel about these choices?